index.js

```diff
@@ -106,24 +105,60 @@ BroccoliMergeTrees.prototype._applyPatch = function (patch) {
     });
 
     switch(operation) {
-      case 'linkdir':   return symlinkOrCopySync(inputFilePath, outputFilePath);
-      case 'mkdir':     return fs.mkdirSync(outputFilePath);
-      case 'rmdir':     return fs.rmdirSync(outputFilePath);
+      case 'mkdir':     return this._applyMkdir(entry, inputFilePath, outputFilePath);
+      case 'rmdir':     return this._applyRmdir(entry, inputFilePath, outputFilePath);
       case 'unlink':    return fs.unlinkSync(outputFilePath);
-      case 'unlinkdir': return unlinkOrRmrfSync(outputFilePath);
       case 'create':    return symlinkOrCopySync(inputFilePath, outputFilePath);
-      case 'change':
-        fs.unlinkSync(outputFilePath);
-        return symlinkOrCopySync(inputFilePath, outputFilePath);
+      case 'change':    return this._applyChange(entry, inputFilePath, outputFilePath);
     }
 
   }, this);
 };
 
+BroccoliMergeTrees.prototype._applyMkdir = function (entry, inputFilePath, outputFilePath) {
+  if (entry.linkDir) {
+    return symlinkOrCopySync(inputFilePath, outputFilePath);
+  } else {
+    return fs.mkdirSync(outputFilePath);
+  }
+}
+
+BroccoliMergeTrees.prototype._applyRmdir = function (entry, inputFilePath, outputFilePath) {
+  if (entry.linkDir) {
+    return unlinkOrRmrfSync(outputFilePath);
+  } else {
+    return fs.rmdirSync(outputFilePath);
+  }
+}
+
+BroccoliMergeTrees.prototype._applyChange = function (entry, inputFilePath, outputFilePath) {
+  if (entry.isDirectory()) {
+    if (entry.linkDir) {
+      // directory copied -> link
+      fs.rmdirSync(outputFilePath);
```

Comment:
`change` operations don't happen for file -> directory.  That case produces `[unlink, file], [mkdir dir]`.  This is because they will have different `relativePath`s as the directory one will end with a `/`.